### PR TITLE
[neox-2.x] apply mpt to storages and persist roots

### PR DESCRIPTION
**Pull request by KickSeason**
### Step 4/7  from `neo-ngd/neox` to `neox-2.x`
* apply mpt to storages
* persist state root

@shargon  Please help me review this.

### CODE REVIEW

File: neo/Persistence/LevelDB/DbTrieStore.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+namespace Neo.Persistence.LevelDB
+{
+    public class DbTrieStore : IKVStore
+    {
+        private readonly DB db;
+        private readonly ReadOptions options;
+        private readonly WriteBatch batch;
+        private readonly byte prefix;
+
+        private readonly byte[] ROOT_KEY = Encoding.ASCII.GetBytes("CURRENT_ROOT");
```
Comment (shargon) [inline neo/Persistence/LevelDB/DbTrieStore.cs:16]: Genesis hash block better?

File: neo/Persistence/LevelDB/DbTrieStore.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        public DbTrieStore(DB db, ReadOptions options, WriteBatch batch, byte prefix)
+        {
+            this.db = db;
+            this.options = options;
```
Comment (shargon) [inline neo/Persistence/LevelDB/DbTrieStore.cs:21]:
```suggestion
            this.options = options ?? ReadOptions.Default;
```

**Comment by KickSeason** [comment]
@shargon Can we merge?

**Comment by KickSeason** [comment]
@erikzhang Can we merge this?

**Comment by erikzhang** [comment]
Once you got at least one approval from the core developers, you can merge it.

**Comment by KickSeason** [comment]
@shargon Please merge this.

**Pull request merged**